core/src/main/java/org/mule/runtime/core/privileged/processor/chain/AbstractMessageProcessorChain.java

```diff
@@ -122,8 +122,11 @@ private ReactiveProcessor applyInterceptors(List<BiFunction<Processor, ReactiveP
 
     // #1 Update MessagingException with failing processor if required, create Error and set error context.
     interceptors.add((processor, next) -> stream -> from(stream)
-        .transform(next)
-        .onErrorMap(MessagingException.class, resolveMessagingException(processor)));
+        .concatMap(event -> just(event)
+            .transform(next)
+            .onErrorMap(MessagingException.class, resolveMessagingException(processor))
```

Comment:
These need to go the other way around don't they?

Reply:
i put then in the other order, won;t the messaging exception be built twice?